vcsclient/factory.go

```diff
@@ -37,8 +35,13 @@ func (builder *ClientBuilder) Token(token string) *ClientBuilder {
 }
 
 // Logger sets the logger
-func (builder *ClientBuilder) Logger(logger *log.Logger) *ClientBuilder {
-	builder.logger = logger
+func (builder *ClientBuilder) Logger(logger Log) *ClientBuilder {
+	if logger != nil {
+		builder.logger = logger
+		return builder
+	}
+
+	builder.logger = EmptyLogger{}
 	return builder
```

Comment:
I think it is reasonable to assume that the input logger is not null.
```suggestion
        builder.logger = logger
	return builder
```

Reply:
done